portal-web/test/functional/com/liferay/portalweb/macros/SearchExperiences.macro

```diff
@@ -0,0 +1,238 @@
+			value1 = "${uiConfigurationJSON}");
+
+		PortletEntry.save();
+	}
+
+	macro addQueryElements {
+		AssertElementPresent(locator1 = "//div[contains(@class,'sidebar-header')]//span[contains(.,'Add Query Elements')]");
+
+		Type.sendKeysApplicationSearch(
+			locator1 = "SearchExperiences#BLUEPRINTS_FRAGMENT_QUERY_SEARCH_BAR",
+			value1 = "${fragment}");
+
+		MouseOver(
+			key_fragmentName = "${fragment}",
+			locator1 = "SearchExperiences#BLUEPRINTS_QUERY_FRAGMENT_NAME");
+
+		Click.clickNoMouseOverNoWaitForVisible(
+			key_fragmentName = "${fragment}",
+			locator1 = "SearchExperiences#BLUEPRINTS_ADD_QUERY_FRAGMENT");
+
+		if (isSet(assetType)) {
+			Click(locator1 = "SearchExperiences#BLUEPRINTS_QUERY_BUILDER_ASSET_TYPE_DROPDOWN");
+
+			Click(
+				locator1 = "SearchExperiences#BLUEPRINTS_QUERY_BUILDER_ASSET_TYPE_OPTION",
+				sortOption = "${assetType}");
+		}
+
+		if (isSet(elasticsearchQuery)) {
+			Type.typeCodeMirrorEditor(
+				locator1 = "SearchExperiences#BLUEPRINTS_ELASTICSEARCH_QUERY_JSON",
+				value1 = "${elasticsearchQuery}");
+		}
+
+		if (isSet(numberBoost)) {
+			Type(
+				key_fieldLabel = "Boost",
+				locator1 = "SearchExperiences#BLUEPRINTS_QUERY_BUILDER_NUMBER_FIELD",
+				value1 = "${numberBoost}");
+		}
+
+		if (isSet(tag1)) {
+			Type(
+				key_fieldLabel = "Tag 1",
+				locator1 = "SearchExperiences#BLUEPRINTS_QUERY_TEXT_FIELD",
+				value1 = "${tag1}");
+		}
+
+		if (isSet(tag2)) {
+			Type(
+				key_fieldLabel = "Tag 2",
+				locator1 = "SearchExperiences#BLUEPRINTS_QUERY_TEXT_FIELD",
+				value1 = "${tag2}");
+		}
+
+		PortletEntry.save();
+	}
+
+	macro deleteBlueprint {
+		Navigator.gotoNavItem(navItem = "Blueprints");
+
+		LexiconEntry.delete(rowEntry = "${name}");
+	}
+
+	macro editBlueprint {
+		Navigator.gotoNavItem(navItem = "Blueprints");
+
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "Edit",
+			rowEntry = "${name}");
+	}
+
+	macro editElement {
+		SearchExperiences.openSearchExperiences(siteURLKey = "guest");
+
+		Navigator.gotoNavItem(navItem = "Elements");
+
+		LexiconEntry.gotoEntryMenuItem(
+			menuItem = "Edit",
+			rowEntry = "${name}");
+	}
+
+	macro openSearchExperiences {
+		Navigator.openWithAppendToBaseURL(urlAppend = "group/${siteURLKey}/~/control_panel/manage?p_p_id=com_liferay_portal_search_tuning_blueprints_web_internal_portlet_BlueprintsAdminPortlet");
+	}
+
+	macro searchBlueprintsPreview {
+		Type(
+			locator1 = "SearchExperiences#BLUEPRINTS_PREVIEW_SEARCH_BAR",
+			value1 = "${searchTerm}");
+
+		var browserType = PropsUtil.get("browser.type");
+
+		if ("${browserType}" == "safari") {
+			Click(locator1 = "Icon#BASIC_SEARCH");
+		}
+		else {
+			KeyPress(
+				locator1 = "SearchExperiences#BLUEPRINTS_PREVIEW_SEARCH_BAR",
+				value1 = "\RETURN");
+		}
+
+		WaitForLiferayEvent.initializeLiferayEventLog();
+	}
+
+	macro searchReactiveSearchWidget {
+		AssertElementPresent(locator1 = "SearchExperiences#BLUEPRINTS_SEARCH_BAR");
```

Comment:
Good point, it's pretty confusing right now. What if we prefixed all Blueprints Admin paths with `BLUEPRINTS_ADMIN`?

Reply:
yeah I like that idea of namespacing paths that are specific to that landing page. I think using `ADMIN` is still ambiguous tho since technically when you're editing an individual blueprint you're still in the admin portlet. so maybe `LANDING` or `HOME` would work?

Reply:
Before we were using this macro to search with the Search Blueprints Widget and in the Blueprints admin page but it is no longer being used with the widget so I think this only works with the blueprints admin search bar, so renaming it makes sense!